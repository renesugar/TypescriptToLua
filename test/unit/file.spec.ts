import * as util from "../util";

describe("JSON", () => {
    test.each([0, "", [], [1, "2", []], { a: "b" }, { a: { b: "c" } }])("JSON (%p)", json => {
        util.testModule(JSON.stringify(json)).setMainFileName("main.json").expectToEqual(json);
    });

    test("empty file throws runtime error", () => {
        util.testModule("")
            .setMainFileName("main.json")
            .expectToEqual(new util.ExecutionError("Unexpected end of JSON input"));
    });
});

describe("shebang", () => {
    test("LF", () => {
        util.testModule`#!/usr/bin/env lua\n
            const foo = true;
        `.expectLuaToMatchSnapshot();
    });

    test("CRLF", () => {
        util.testModule`#!/usr/bin/env lua\r\n
            const foo = true;
        `.expectLuaToMatchSnapshot();
    });
});
