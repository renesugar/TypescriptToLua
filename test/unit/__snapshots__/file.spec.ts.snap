// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shebang CRLF 1`] = `
"#!/usr/bin/env lua
foo = true"
`;

exports[`shebang LF 1`] = `
"#!/usr/bin/env lua
foo = true"
`;
